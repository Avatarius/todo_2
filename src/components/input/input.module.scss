.form {
  margin: 0;
  padding: 0;
}

.input {
  inline-size: 100%;
  display: block;
  box-sizing: border-box;
  border: none;
  border-block-end: 1px solid var(--border-color, #77777780);
  border-start-start-radius: var(--border-radius, 5px);
  border-start-end-radius: var(--border-radius, 5px);
  padding: 7px;
  font-family: inherit;
  font-size: 32px;
  appearance: none;
}
