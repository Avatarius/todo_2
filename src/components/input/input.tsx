
import { ChangeEvent, Dispatch, FormEvent, SetStateAction } from 'react';
import styles from './input.module.scss';
import { ITodo } from '../../utils/types';

interface IInputProps {
  newTodo: string;
  setNewTodo: Dispatch<SetStateAction<string>>;
  addTodo: () => void;

}

function Input(props: IInputProps) {
  const {newTodo, setNewTodo, addTodo} = props;

  function handleSubmit(e: FormEvent) {
    e.preventDefault();
    if (!newTodo) return;
    addTodo();
    setNewTodo('');
  }

  function handleChange(e: ChangeEvent<HTMLInputElement>) {
    setNewTodo(e.target.value);
  }


  return (
    <form className={styles.form} onSubmit={handleSubmit} data-testid='form'>
      <input type="text" className={styles.input} placeholder='What needs to be done?' value={newTodo} onChange={handleChange} data-testid='input'/>
    </form>
  )
}


export {Input};
