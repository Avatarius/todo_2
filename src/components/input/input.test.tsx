import { render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import { expect, test, describe, jest } from "@jest/globals";
import { Input } from "./input";

describe("Компонент <Input/>", () => {
  test("Рендер компонента", () => {
    const component = renderer.create(
      <Input newTodo="test" addTodo={() => {}} setNewTodo={() => {}} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("Submit", () => {
    const setTodo = jest.fn();
    render(<Input newTodo="fdfs" addTodo={() => {}} setNewTodo={setTodo} />);
    const form = screen.getByTestId("form");
    expect(form).toBeTruthy();
    userEvent.type(screen.getByTestId("input"), "test");
    fireEvent.submit(form);
    expect(setTodo).toHaveBeenCalledWith("");
  });
});
