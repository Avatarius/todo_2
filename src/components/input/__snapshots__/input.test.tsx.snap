// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент <Input/> Рендер компонента 1`] = `
<form
  data-testid="form"
  onSubmit={[Function]}
>
  <input
    data-testid="input"
    onChange={[Function]}
    placeholder="What needs to be done?"
    type="text"
    value="test"
  />
</form>
`;
